packages/flutter/lib/src/services/raw_keyboard_macos.dart

```diff
@@ -0,0 +1,227 @@
+  /// various key combinations with Unicode characters. This property is only
+  /// valid for key-up and key-down events. It raises an
+  /// NSInternalInconsistencyException if accessed on any other kind of event object.
+  ///
+  /// This property is set to an empty string for dead keys, such as Option-e.
+  /// However, for a key combination such as Option-Shift-e this property is set
+  /// to the standard accent ("´").
+  final String characters;
+
+  /// The characters generated by a key event as if no modifier key (except for
+  /// Shift) applies.
+  ///
+  /// This property is set to the non-modifier key character pressed for dead
+  /// keys, such as Option-e. For example, Option-e (no shift key) returns an
+  /// “e" for this method, whereas the characters property returns an empty string.
+  ///
+  /// This property is useful for determining “basic” key values in a
+  /// hardware-independent manner, enabling such features as keyboard equivalents
+  /// defined in terms of modifier keys plus character keys. For example, to
+  /// determine if the user typed Alt-S, you don’t have to know whether Alt-S
+  /// generates a German double ess, an integral sign, or a section symbol. You
+  /// simply examine the string contained by this property along with the event’s
+  /// modifier flags, checking for “s” and NSAlternateKeyMask.
+  final String charactersIgnoringModifiers;
+
+  /// The virtual key code for the keyboard key associated with a key event.
+  ///
+  /// The property’s value is hardware-independent. The value returned is the
+  /// same as the value returned in the kEventParamKeyCode when using Carbon
+  /// Events.
+  final int keyCode;
+
+  /// A mask of the current modifiers using the values in Modifier Flags.
+
+  /// This returns the state of devices combined with synthesized events at
+  /// the moment, independent of which events have been delivered via the
+  /// event stream.
+  final int modifiers;
+
+  @override
+  String get keyLabel => charactersIgnoringModifiers.isEmpty ? null : charactersIgnoringModifiers;
+
+  @override
+  PhysicalKeyboardKey get physicalKey => kMacOsToPhysicalKey[keyCode] ?? PhysicalKeyboardKey.none;
+
+  @override
+  LogicalKeyboardKey get logicalKey {
+    // Look to see if the keyCode is a printable number pad key, so that a
+    // difference between regular keys (e.g. "=") and the number pad version
+    // (e.g. the "=" on the number pad) can be determined.
+    final LogicalKeyboardKey numPadKey = kMacOsNumPadMap[keyCode];
+    if (numPadKey != null) {
+      return numPadKey;
+    }
+
+    // Look to see if the keyCode is one we know about and have a mapping for.
+    if (keyLabel != null &&
+        !LogicalKeyboardKey.isControlCharacter(keyLabel) &&
+        charactersIgnoringModifiers.isNotEmpty) {
+      final int keyId = LogicalKeyboardKey.unicodePlane |
+        (charactersIgnoringModifiers.codeUnitAt(0) & LogicalKeyboardKey.valueMask);
```

Comment:
You might put in a comment/todo/assert addressing the case where `charactersIgnoringMultipliers` might have more than one code point.  You could also try and handle up to two code points by shifting one up by 16 bits and or'ing them together. I didn't need to do that on Android, since the platform only supplies one code point as an int.

Reply:
Looked a bit more online and seems unlikely, although not impossible to receive more than 1. I handled 2 code points as you suggested.

Reply:
Yeah, I agree, it's unlikely.